datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -581,45 +502,118 @@ pub fn array_except(args: &[ArrayRef]) -> Result<ArrayRef> {
 
 pub fn array_slice(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
Also here I think it would help to add a comment explaining what the arguments are to this function and what the expected output is

Key details might be if the index values are 1 based or 0 based, for example